WordPress/gutenberg
# Block Editor: LinkControl: Add Submit button as action

Extracted from #19462 
Related: #19570

This pull request seeks to add a new "Submit" button to be shown ~alongside~ instead of the existing "Reset" button in the new `LinkControl` component used by the Navigation and Buttons blocks.

Before|After
---|---
![Before](https://user-images.githubusercontent.com/1779930/73485531-3eab4a80-4371-11ea-9e02-ca247d2c9870.png)|![After](https://user-images.githubusercontent.com/1779930/73700760-696d0a00-46b5-11ea-9b2d-0a58ac35efc5.png)

<details><summary>Original Before/After</summary>
<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/1779930/73485531-3eab4a80-4371-11ea-9e02-ca247d2c9870.png"><img src="https://user-images.githubusercontent.com/1779930/73485531-3eab4a80-4371-11ea-9e02-ca247d2c9870.png" alt="Before" style="max-width:100%;"></a></td>
<td><a target="_blank" rel="noopener noreferrer" href="https://user-images.githubusercontent.com/1779930/73484871-18d17600-4370-11ea-984a-5249a3edd53a.png"><img src="https://user-images.githubusercontent.com/1779930/73484871-18d17600-4370-11ea-984a-5249a3edd53a.png" alt="After" style="max-width:100%;"></a></td>
</tr>
</tbody>
</table>
</details>

**Rationale**

It only really occurs to me now, but I'm not really sure what benefits this brings.

If I had to guess:

- It could make it visually clearer that this is a form to be submitted, whereas otherwise it relies on the user having knowledge that pressing <kbd>Enter</kbd> is an equally effective means to submit the form.
- According to the original commit 90f7b5dc4a42f2403559eb28e396dbb830d42042 from #19462, "When clicking Change on an existing link you should be able to modify and then resubmit the link without having to repeat your search over again.". It's not really clear to me what impact there is on this behavior in whether or not a Submit button is present, though this could have been more relevant prior to changes in the form submission behavior as in #19651 (cc @getdave).

Even if we decide not to move forward with adding a submit button, and depending on some decision out of #19570 as to whether the "Reset" button should be removed as well, the proposed changes could still be valuable in providing a more reliable computation of offsets for absolute positioning of the spinner and reset buttons.

**Testing Instructions:**

Verify that when editing a link in a Navigation or Buttons block, you should see a Submit button shown an activated as long as the input contains a valid link.

Check as well that issues raised in #19462 with the original implementation have been addressed:

- https://github.com/WordPress/gutenberg/pull/19462#issuecomment-579824527
- https://github.com/WordPress/gutenberg/pull/19462#issuecomment-579481208
- #19544
